src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1294,6 +1294,11 @@ internal static object[] GetCustomAttributes(RuntimeType type, RuntimeType caTyp
                 type = type.BaseType as RuntimeType;
             }
```

Comment:
You can replace `List<object>` with `ListBuilder<T>` to save another allocation.

Reply:
Not sure I follow this note - `ListBuilder<T>` is private inside `RuntimeType` and expects a known capacity at start, do you want to consider changing either of those things? Or aiming at `ValueListBuilder<T>`?

Reply:
Default `ListBuilder<T>` constructor does not need capacity, and changing it to internal should be fine too.

`ValueListBuilder` would work too, but probably not as well as `ListBuilder<T>` in this case.

Reply:
Got it - I'm still figuring out what scope of changes is acceptable in which areas of CoreFX/CoreCLR, so I really appreciate the feedback! `ListBuilder<T>` is exposed/used now, and I've greatly expanded the benchmarks above to illustrate the gains there. Thanks!